# win10 设定计划任务时提示所指定的账户名称无效，如何解决？

我想把我的 python 爬虫脚本设定为自动定时执行，我的设备是win10 操作系统，这将用到系统自带的计划任务功能。且我希望不管用户是否登录都要运行该定时任务，但在设置计划任务的属性时，遇到一个报错：**所指定的账户名称无效**。

该报错是如何发生的，以及如何解决？记录如下：

## 报错是如何发生的？

如下图所示，设置计划任务的属性：
- 如果仅勾选“只在用户登录时运行”，点击“确定”后直接创建成功。并不会遇到如题报错。

- 如果勾选“**不管用户是否登录都要运行**”，点击“确定”后弹出账户信息输入界面。这种情况才会遇到这个错误。

![image](https://user-images.githubusercontent.com/31027645/64668533-7e3fcb80-d490-11e9-99eb-e523f58b7891.png)


这种情况下，正确输入密码，也会遇到报错：所指定的账户名称无效

![image](https://user-images.githubusercontent.com/31027645/64668542-8b5cba80-d490-11e9-8617-56dffe679420.png)


报错截图：任务 name 出错。错误消息：所指定的账户名称无效。

![image](https://user-images.githubusercontent.com/31027645/64668550-944d8c00-d490-11e9-90dd-afafecdaec76.png)


## 如何解决该报错？
我搜了好多办法，唯一对我的情况有效的解决办法是：在计划任务属性页面，点击“更改用户或组”，输入用户名（比如我的“75801”），然后点击“检查名称”，再点击“确定”提交，就可以了。

**请留意，执行该操作前与之后，“安全选项-运行该任务时，请使用下列用户账户”那里发生了变化。**

![image](https://user-images.githubusercontent.com/31027645/64668567-a5969880-d490-11e9-92cc-47201257c054.png)

![image](https://user-images.githubusercontent.com/31027645/64668576-adeed380-d490-11e9-994e-3056138e6533.png)

无论是设定计划任务，还是修改计划任务的属性，当勾选了“不管用户是否登录都要运行”后，都有可能遇到此类报错。都可以按照这个方式解决问题。
